modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/action/executor/test/ObjectActionExecutorRegistryImplTest.java

```diff
@@ -0,0 +1,143 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.object.action.executor.test;
+
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.object.action.executor.ObjectActionExecutor;
+import com.liferay.object.action.executor.ObjectActionExecutorRegistry;
+import com.liferay.object.scope.CompanyScoped;
+import com.liferay.object.scope.ObjectDefinitionScoped;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.kernel.util.UnicodeProperties;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceRegistration;
+
+/**
+ * @author Guilherme Camacho
+ */
+@RunWith(Arquillian.class)
+public class ObjectActionExecutorRegistryImplTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Test
+	public void testShouldNotReturnObjectActionExecutorScopedByAnotherCompany() {
+		ServiceRegistration<ObjectActionExecutor>
+			objectActionExecutorServiceRegistration1 = null;
+
+		ServiceRegistration<ObjectActionExecutor>
+			objectActionExecutorServiceRegistration2 = null;
+
+		try {
+			objectActionExecutorServiceRegistration1 =
+				_registerObjectActionExecutor(
+					new TestObjectActionExecutor(1, Collections.emptyList()));
+
+			ObjectActionExecutor testObjectActionExecutor =
```

Comment:
@guilhermedcamacho I believe we should rename to `objectActionExecutor` because the type of the variable is ObjectActionExecutor, I believe this name only makes sense if it you explicit define the implementation TestObjectActionExecutor as the type,

makes sense?

Thanks!